### PR TITLE
osd/scrub: fix the handling of deep-scrub when noscrub is set

### PR DESCRIPTION
**Pull request by ronen-fr**
Recent scrub scheduling code errs in (at one location) incorrectly considering noscrub as not
precluding deep-scrub.

Fixes: https://tracker.ceph.com/issues/52901
Signed-off-by: Ronen Friedman <rfriedma@redhat.com>

### CODE REVIEW

File: src/osd/PG.cc [hunk 1]
```diff
@@ -1478,15 +1478,20 @@ bool PG::verify_periodic_scrub_mode(bool allow_deep_scrub,
...
-  bool try_to_auto_repair =
-    (cct->_conf->osd_scrub_auto_repair && get_pgbackend()->auto_repair_supported());
+  bool try_to_auto_repair = (cct->_conf->osd_scrub_auto_repair &&
+                             get_pgbackend()->auto_repair_supported());
+
+  dout(10) << __func__ << " pg: " << info.pgid
+           << " allow: " << allow_regular_scrub << "/" << allow_deep_scrub
+           << " deep errs: " << has_deep_errors
+           << " auto-repair: " << try_to_auto_repair << " ("
+           << cct->_conf->osd_scrub_auto_repair << ")" << dendl;
```
[the remark this answers is not in the document]
Comment (ljflores) [reply src/osd/PG.cc:1494]: Makes sense

**Comment by ronen-fr** [comment]
@neha-ojha 
> Was https://tracker.ceph.com/issues/52901 discovered by a test failure?
> Since this a bug fix, it will be useful to add details about how the bug was discovered in the tracker and if it was discovered by a test failure, add a test run to reproduce it and another (with the fix) to demonstrate the patch fixes the issue.

Originally - the problem was discovered on my desk. But once I noticed it - I think it's the failure behind some other test failures, e.g. http://pulpito.front.sepia.ceph.com/yuriw-2021-11-03_15:48:02-rados-wip-yuri7-testing-2021-11-01-1748-distro-basic-smithi/

I will add a few lines to the bug description. Note - it's a nuisance that creates hard-to-understand failures, and should be merged as soon as possible.

**Comment by neha-ojha** [comment]
> @neha-ojha
> 
> > Was https://tracker.ceph.com/issues/52901 discovered by a test failure?
> > Since this a bug fix, it will be useful to add details about how the bug was discovered in the tracker and if it was discovered by a test failure, add a test run to reproduce it and another (with the fix) to demonstrate the patch fixes the issue.
> 
> Originally - the problem was discovered on my desk. But once I noticed it - I think it's the failure behind some other test failures, e.g. http://pulpito.front.sepia.ceph.com/yuriw-2021-11-03_15:48:02-rados-wip-yuri7-testing-2021-11-01-1748-distro-basic-smithi/
> 
> I will add a few lines to the bug description. Note - it's a nuisance that creates hard-to-understand failures, and should be merged as soon as possible.

sounds good to me!

**Comment by ronen-fr** [comment]
jenkins retest this please

**Pull request merged**